deepchem/feat/base_classes.py

```diff
@@ -267,17 +267,23 @@ def featurize(self, molecules, log_every_n=1000):
     for i, mol in enumerate(molecules):
       if i % log_every_n == 0:
         logger.info("Featurizing datapoint %i" % i)
+
       try:
         if isinstance(mol, str):
           # mol must be a RDKit Mol object, so parse a SMILES
           mol = Chem.MolFromSmiles(mol)
           # SMILES is unique, so set a canonical order of atoms
           new_order = rdmolfiles.CanonicalRankAtoms(mol)
           mol = rdmolops.RenumberAtoms(mol, new_order)
+
         features.append(self._featurize(mol))
-      except:
+      except Exception as e:
+        if isinstance(mol, Chem.rdchem.Mol):
+          mol = Chem.MolToSmiles(mol)
         logger.warning(
-            "Failed to featurize datapoint %d. Appending empty array", i)
+            "Failed to featurize datapoint %d, %s. Appending empty array", i,
+            mol)
+        logger.warning("Exception message: {}".format(e))
         features.append(np.array([]))
```

Comment:
I added more logs. I seems that this is noisy for someone...

Reply:
The Tox21 logs

```
Failed to featurize datapoint 95, [I-].[K+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 255, [Hg+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 659, [Ba+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 985, [TlH2+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1423, [Cr+3]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1534, [Fe+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1722, [Co+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1933, [PbH2+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2147, [Fe+3]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2251, [Cu+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2760, [Cd+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2832, [SnH2+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4024, [Mn+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4375, [Be+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4611, [Zn+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 5942, [Br-].[Na+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
```

Reply:
I think this is useful! It's clear from this what we're failing on :)